java-frontend/src/main/java/org/sonar/java/se/LearnedConstraint.java

```diff
@@ -22,16 +22,11 @@
 import org.sonar.java.se.constraint.Constraint;
 import org.sonar.java.se.symbolicvalues.SymbolicValue;
 
-import javax.annotation.CheckForNull;
-import javax.annotation.Nullable;
-
 public class LearnedConstraint {
   final SymbolicValue sv;
-
-  @Nullable
   final Constraint constraint;
 
-  public LearnedConstraint(SymbolicValue sv, @Nullable Constraint constraint) {
+  public LearnedConstraint(SymbolicValue sv, Constraint constraint) {
     this.sv = sv;
```

Comment:
Maybe adding a `Preconditions.checkNotNull(constraint)` test in here, as we are changing the behavior, to be sure that no further null constraints are going to be created.